File: eGlamHeelHangout/eGlamHeelHangout.Service/Database/Giveaway.cs
using System;
using System.Collections.Generic;

namespace eGlamHeelHangout.Service.Database
{
    public partial class Giveaway
    {
        public int GiveawayId { get; set; }

        public DateTime EndDate { get; set; }
        public string Color { get; set; }
        public string Description { get; set; }
        public decimal HeelHeight { get; set; }
        public bool IsClosed { get; set; } = false;
        public string Title { get; set; }
        public byte[] GiveawayProductImage { get; set; }
        public virtual ICollection<GiveawayParticipant> GiveawayParticipants { get; set; } = new List<GiveawayParticipant>();
    }
}

File: eGlamHeelHangout/UI/eglamheelhangout_admin/lib/models/giveaway.dart
import 'package:json_annotation/json_annotation.dart';

part 'giveaway.g.dart'; 

@JsonSerializable()
class Giveaway {

    int? giveawayId;
    String title;
    String color;
    double heelHeight;
    String description;
    DateTime endDate;
    String giveawayProductImage;
    String? winnerName; 
    bool isClosed;
    int? participantsCount;           
    bool? canGenerateWinner;          
    bool? canDelete;                 




    Giveaway({
    this.giveawayId,
    required this.title,
    required this.color,
    required this.heelHeight,
    required this.description,
    required this.endDate,
    required this.giveawayProductImage,
    this.winnerName,
    required this.isClosed,
    this.participantsCount,
    this.canGenerateWinner,
    this.canDelete,

     
});

    factory Giveaway.fromJson(Map<String, dynamic> json) => _$GiveawayFromJson(json);
    
    Map<String, dynamic> toJson() => _$GiveawayToJson(this);
}